ruby/ql/lib/codeql/ruby/controlflow/internal/ControlFlowGraphImpl.qll

```diff
@@ -416,17 +399,31 @@ module Trees {
         first(this.getBranch(i + 1), succ) and
         c.(ConditionalCompletion).getValue() = false
       )
+      or
+      succ = this and
```

Comment:
Don't we need a `c instanceof NormalCompletion` check here?

Reply:
don't we get that "for free" because `value` and `branch(_)` are mentioned in `propagatesAbnormal`?

Reply:
`propagatesAbnormal` only gives you the escaping `raise` edge (typically to the exit point), it doesn't prevent an internal edge.